src/core/image.rs

```diff
@@ -96,6 +96,10 @@ where
     fn entrypoint(&self) -> Option<String> {
         None
     }
+
+    fn expose_ports(&self) -> Option<Vec<u16>> {
+        None
+    }
```

Comment:
Are we gaining anything from wrapping the `Vec` in an `Option`?

Reply:
Gotcha. I think we can go with zero-sized `Vec` for default cases.

Reply:
I think that should work yes.